openstack/blockstorage/v2/snapshots/results.go

```diff
@@ -0,0 +1,98 @@
+package snapshots
+
+import (
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/pagination"
+)
+
+// Snapshot contains all the information associated with a Cinder Snapshot.
+type Snapshot struct {
+	// Unique identifier.
+	ID string `json:"id"`
+
+	// Date created.
+	CreatedAt gophercloud.JSONRFC3339MilliNoZ `json:"created_at"`
```

Comment:
@jrperritt [v1 snapshots](https://github.com/gophercloud/gophercloud/blob/master/openstack/blockstorage/v1/snapshots/results.go) (and v1 volumes) aren't set to `time.Time` -- was that missed during the "time" update? I can't think of a reason that they would follow a different pattern.

@j-griffith If `time.Time` should be used, you'll want to set `CreatedAt` and `UpdatedAt` to `time.Time` here and then do the conversion to `gophercloud.JSONRFC3339MilliNoZ` in an `UnmarshalJSON` function like [here](https://github.com/gophercloud/gophercloud/blob/master/openstack/blockstorage/v2/volumes/results.go#L80-L97).

The unit tests will also need updated like [here](https://github.com/gophercloud/gophercloud/blob/master/openstack/blockstorage/v2/volumes/testing/requests_test.go#L69).